### PR TITLE
refine checkout metrics

### PR DESCRIPTION
**Pull request by joelhooks**
Cleaning up and making the funnel for checkout more detailed

### CODE REVIEW

File: src/api/stripe/stripe-checkout-redirect.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ const stripeCheckoutRedirect = async (options: {
       site: 'egghead.io',
       client_id: process.env.NEXT_PUBLIC_CLIENT_ID,
       success_url: `${process.env.NEXT_PUBLIC_DEPLOYMENT_URL}/confirm/membership?session_id={CHECKOUT_SESSION_ID}`,
-      cancel_url: `${process.env.NEXT_PUBLIC_DEPLOYMENT_URL}/pricing`,
+      cancel_url: `${process.env.NEXT_PUBLIC_DEPLOYMENT_URL}/pricing?stripe=cancelled`,
```
Comment (joelhooks) [inline src/api/stripe/stripe-checkout-redirect.ts:37]: this will note that they are canceling the action

File: src/pages/confirm/membership.tsx [hunk 2]
```diff
@@ -23,6 +24,9 @@ const ConfirmMembershipPage: React.FC = () => {
         .get(`/api/stripe/checkout/session?session_id=${session_id}`)
         .then(({data}) => {
           setSession(data)
+          track('checkout: membership confirmed', {
```
Comment (joelhooks) [inline src/pages/confirm/membership.tsx:27]: we weren't tracking ultimate success with a conversion

File: src/pages/pricing/email/index.tsx [hunk 3]
```diff
@@ -49,6 +49,9 @@ const Email: React.FunctionComponent<EmailFormProps> & {getLayout: any} = ({
       setIsError(
         `You've already got a pro account at ${email}. [Please login](/login).`,
       )
+      track('checkout: existing pro account found', {
```
Comment (joelhooks) [inline src/pages/pricing/email/index.tsx:52]: this is a login fail state if we match the email to an existing pro account

File: src/pages/pricing/index.tsx [hunk 4]
```diff
@@ -30,6 +30,13 @@ const Pricing: FunctionComponent<PricingProps> & {getLayout: any} = () => {
   const [priceId, setPriceId] = React.useState<string>()
   const router = useRouter()
 
+  React.useEffect(() => {
+    track('visited pricing')
```
Comment (joelhooks) [inline src/pages/pricing/index.tsx:34]: we weren't tracking a page visit so now we are

File: src/pages/pricing/index.tsx [hunk 5]
```diff
@@ -30,6 +30,13 @@ const Pricing: FunctionComponent<PricingProps> & {getLayout: any} = () => {
...
+    if (router?.query?.stripe === 'cancelled') {
+      track('checkout: cancelled from stripe')
```
Comment (joelhooks) [inline src/pages/pricing/index.tsx:36]: if the page visit has a query for 'cancelled' we track that

**Review by dealingwith — APPROVED** [review]

**Pull request merged**